Time of the day II (Issue/8419)

kingfisher88 opened the pull request.

kingfisher88 committed:
Time of the day II

Create a simple admin page for the time of day service
Set time of day intervals

kingfisher88 committed:
PHP code checking with Code Sniffer

kingfisher88 committed:
Use Dependency Injection in GreetingTime.php

kingfisher88 committed:
Prettier for SCSS

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8419

gkurucz commented on web/modules/custom/usergreeting/src/Form/UsergreetingConfigForm.php:
```diff
@@ -0,0 +1,107 @@
+/**
+ * Class UsergreetingConfigForm.
+ */
+class UsergreetingConfigForm extends ConfigFormBase {
```
Make this class final if it's not extended further.

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8419

gkurucz commented on web/modules/custom/usergreeting/src/Form/UsergreetingConfigForm.php:
```diff
@@ -0,0 +1,107 @@
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['usegreeting.settings'];
```
Are you sure this is the right name? Shouldn't this be `usergreeting`? :smile: 

This is probably a typo that should be fixed everywhere.

gkurucz commented on web/modules/custom/usergreeting/src/Form/UsergreetingConfigForm.php:
```diff
@@ -0,0 +1,107 @@
+    $afternoon = $form_state->getValue('afernoon_start');
+    $evening = $form_state->getValue('evening_start');
+
+    if ($morning == $afternoon) {
```
Please use strict comparison operator. `===` instead of `==`.

gkurucz commented on web/modules/custom/usergreeting/config/install/usergreeting.settings.yml:
```diff
@@ -0,0 +1,4 @@
+usergreeting_defaults:
```
Besides providing a default settings file, it's also important to provide the schema for the stored data.

You can find examples in:

`web/core/modules/serialization/config/schema/serialization.schema.yml`
`web/core/modules/action/config/schema/action.schema.yml`

And there's a [comprehensive guide](https://www.drupal.org/docs/8/api/configuration-api/configuration-schemametadata) about them as well.

Apart from that, well done! :slightly_smiling_face: This PR looks very good.

lekob replied:
The `usergreeting_defaults` key is unnecessary. Drupal will ignore it when dealing with the configuration.

kingfisher88 committed:
Add usergreeting.schema.yml
Change the comparison operator, correct typo  usegreeting to usergreeting,
make class UsergreetingConfigForm to final in UsergreetinConfigForm.php

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8419

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8419

lekob commented on web/modules/custom/usergreeting/config/install/usergreeting.settings.yml:
```diff
@@ -0,0 +1,4 @@
+usergreeting_defaults:
+  morning_start: 5
+  afernoon_start: 12
```
afternoon.

kingfisher88 committed:
Remove unnecessery key and correct typo in usergreeting.settings.yml
Add a single whitespace to greeting message in GreetinTime.php

kingfisher88 committed:
Merge branch 'master' of github.com:Pronovix/food into issue/8419

gkurucz commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -61,15 +72,18 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
-    if ($time_output >= 5 && $time_output < 12) {
-      $greeting = $this->t('Good Morning');
+    if ($time_output >= $config_morning && $time_output < $config_afternoon) {
+      $greeting = implode(' ', [$greeting = $this->t('Good Morning')]);
```
Unfortunately implode doesn't do anything when you use it like this. Firstly, you don't need to set the value twice, it would be enough to pass `[$this->t('Message')]` into the array. However, implode glues together pieceS of an array into a single string, so if you pass an array with a single element it cannot glue that element to itself, it's only useful if the array contains multiple elements.

If you want the final message to look something like "Good morning User!" then you need to do glue together:

1. the message
1. the username
1. the exclamation mark

But do you really need implode? See my other comment.

gkurucz commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -61,15 +72,18 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
+      $greeting = implode(' ', [$greeting = $this->t('Good Evening')]);
     }
 
     return [
```
You could simplify the contents of this render array if you use `#theme => 'username'` only at the outermost depth. Meaning that you don't need to wrap this into a `html_tag`.

Rather than that, you should use `#prefix` and `#suffix` as seen here: https://www.drupal.org/docs/8/api/render-api/render-arrays#properties

kingfisher88 committed:
Use render arrays: inline_template, prefix and suffix in GreetinTime.php

kingfisher88 committed:
Remove inline_template render array from GreetingTime.php

gkurucz commented on web/modules/custom/usergreeting/src/GreetingTime.php:
```diff
@@ -61,34 +72,33 @@ public function __construct(TimeInterface $timeOutput, AccountInterface $account
         ],
       ],
+      '#prefix' => $greeting,
       'username' => [
```
You don't even need this child element. Simply move the contents of this array one level out.

kingfisher88 committed:
Reorder render elements in return array in GreetingTime.php
